src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1769,11 +1867,26 @@
                                      productions)
                       ;; Store the name of the custom comparator for durability.
                       {:clara.rules.durability/comparator-name `production-load-order-comp})
-        beta-graph (to-beta-graph productions)
-        beta-tree (compile-beta-graph beta-graph)
+
+        ;; A stateful counter used for unique ids of the nodes of the graph.
+        id-counter (atom 0)
+        create-id-fn (fn [] (swap! id-counter inc))
+
+        ;; 1250 is an arbitrary number, this could be lower or higher depending on the
+        ;; rulebase that is being compiled, with regards to the average size of the
+        ;; forms being evaluated.
+        forms-per-eval (:forms-per-eval options 1250)
```

Comment:
It'd be nice to have the 1250 in a var 
`forms-per-eval-default` somewhere and refer to that in all the places that you currently refer to 1250. That way the comments and the compiler vs deserialization impl doesn't get out of sync.

Also, how did you come up with 1250? Is this a threshold you found to work well in practice? It does depend on size of expressions of course.

As an aside to this discussion:
I think ideally, rules would be written in a way where there were not large inline code blocks in the LHS conditions, accumulators, and/or the RHS action forms.

I know from experience that people sometimes tend to make really large RHS forms though (I don't recommend it).

e.g.

Bad
```clj

(defrule my-rule
 
=>
(let [<lots of bindings and logic>]
   (insert! some-result))
```

Good
Good
```clj

(defrule my-rule
 
=>
;; `do-bigger-logic-fn` is evaluated under normal compilation conditions (better).
;; Also, things can be broken up and made more readable by naming functions around them.
;; Also, these pure fn's can be independently verified/tested.
(let [some-result (do-bigger-logic-fn <bindings needed from LHS>)]
   (insert! some-result))
```

If rules were written this way, I'd think the `:forms-per-eval` could be quite a bit higher. I realize that'd be a breaking change for impl's that have abused the RHS by making huge action expressions.

Reply:
@EthanEChristian and I talked a lot about this offline. It actually may be the case that inline forms don't have an effect on the `:forms-per-eval`. This is nice if it proves to be the case.

Basically, the reason is that `batching-try-eval` eval's a vector of expressions of a `fn` form.
e.g.
```clj
[(fn [x] ...) (fn [y] ...) ...]
```

These each get their own classfile, so the big method size ends up just being in the method that creates this vector of `fn` references. There is constant overhead in terms of bytecodes for this method then. That constant overhead can be multiplied by teh `:forms-per-eval` to get a close approx of the correct value to default to. Keeping it configurable is likely still a good idea, due these sizes being possibly variable across Clj compiler versions, etc.

We can make sure to add those details here shortly once it is completely validated.